src/magic_spell_effect.cpp

```diff
@@ -503,3 +509,245 @@ void spell_effect::translocate( const spell &sp, const Creature &caster,
+        num = rng( 1, 3 );
+    } else {
+        caster.add_msg_if_player( m_warning, _( "Giant wasps appear!" ) );
+        bug = static_cast<mtype_id>( "mon_wasp" );
+        num = rng( 1, 2 );
+    }
+    if( bug ) {
+        for( int j = 0; j < num && !empty.empty(); j++ ) {
+            const tripoint spawnp = random_entry_removed( empty );
+            if( monster *const b = g->summon_mon( bug, spawnp ) ) {
+                b->friendly = -1;
+                b->add_effect( static_cast<efftype_id>( "pet" ), 1_turns, num_bp, true );
+            }
+        }
+    }
+}
+
+void spell_effect::light( const spell &sp, const Creature &caster )
+{
+    caster.add_msg_if_player( _( "The %s glows brightly!" ), sp.get_obj_name() );
+    g->events.add( EVENT_ARTIFACT_LIGHT, calendar::turn + 3_minutes );
+}
+
+void spell_effect::growth( const tripoint &target )
+{
+    monster tmptriffid( mtype_id::NULL_ID(), target );
+    mattack::growplants( &tmptriffid );
+}
+
+void spell_effect::mutate( const Creature &caster )
+{
+    player *p = g->critter_at<player>( caster.pos() );
+    if( !one_in( 3 ) ) {
+        p->mutate();
+    }
+}
+
+void spell_effect::teleglow( const Creature &caster )
+{
+    player *p = g->critter_at<player>( caster.pos() );
+    p->add_msg_if_player( m_warning, _( "You feel unhinged." ) );
+    p->add_effect( efftype_id( "effect_teleglow" ), rng( 30_minutes, 120_minutes ) );
+}
+
+void spell_effect::noise( const spell &sp, const Creature &caster, const tripoint &target )
+{
+    sounds::sound( target, sp.damage(), sounds::sound_t::combat,
+                   string_format( _( "a deafening boom from %s %s" ),
+                                  caster.disp_name( true ), sp.get_obj_name() ), true, "misc", "shockwave" );
+}
+
+void spell_effect::scream( const spell &sp, const Creature &caster, const tripoint &target )
+{
+    player *p = g->critter_at<player>( caster.pos() );
+    sounds::sound( target, sp.damage(), sounds::sound_t::alert,
+                   string_format( _( "a disturbing scream from %s %s" ),
+                                  p->disp_name( true ), sp.get_obj_name() ), true, "shout", "scream" );
+    if( !p->is_deaf() ) {
+        p->add_morale( MORALE_SCREAM, -10, 0, 30_minutes, 1_minutes );
+    }
+}
+
+void spell_effect::dim( const Creature &caster )
+{
+    caster.add_msg_if_player( _( "The sky starts to dim." ) );
+    g->events.add( EVENT_DIM, calendar::turn + 5_minutes );
+}
+
+void spell_effect::flash( const Creature &caster, const tripoint &target )
+{
+    caster.add_msg_if_player( _( "The %s flashes brightly!" ), "it->tname()" );
+    explosion_handler::flashbang( target );
+}
+
+void spell_effect::vomit( const Creature &caster )
+{
+    player *p = g->critter_at<player>( caster.pos() );
+    p->add_msg_if_player( m_bad, _( "A wave of nausea passes through you!" ) );
+    p->vomit();
+}
+
+void spell_effect::shadows( const Creature &caster, const tripoint &target )
```

Comment:
i would rather extend the api of the generic summon spell effect than make a brand new one for just shadows to be spawned.